pkg/statemachine/msgbuffers.go

```diff
@@ -35,17 +36,30 @@ func (nbs *nodeBuffers) nodeBuffer(source nodeID) *nodeBuffer {
 			id:       source,
 			logger:   nbs.logger,
 			myConfig: nbs.myConfig,
+			msgBufs:  map[*msgBuffer]struct{}{},
 		}
 	}
 
 	return nb
 }
 
+func (nbs *nodeBuffers) status() []*status.NodeBuffer {
+	var stats []*status.NodeBuffer
+	for _, nb := range nbs.nodeMap {
+		stats = append(stats, nb.status())
+	}
+	return stats
```

Comment:
Because the stats slice was built with map iteration, the entries will be in a random order.  I'd suggest adding a `slice.Sort(` here to ensure that the output of this function is deterministic.